Question: g++ 8 support

Hi,

I have noticed that `g++` version 8.4.0 fails to compile specific example I have, which uses nexus with two different services, which require different invocation signatures. It compiles fine with `clang++` version 8, though.
Do you maybe know if this is a specific `g++` deficiency, or `cib` code could be modified in a way to handle this old g++ version as well?

Compiler error I am seeing is as follows:
```
/repo/github/cib-examples/ticker-extended/src/main.cpp:146:16:   required from here
/repo/github/cib-examples/external/cib/cib/nexus.hpp:50:53: error: invalid conversion from ‘void (*)(const spread_ind&)’ to ‘cib::traits::interface_t<publish_ticker>’ {aka ‘void (*)(std::__cxx11::basic_string<char>, const std::__cxx11::basic_string<char>&)’} [-fpermissive]
                     cib::service<clean_service_tag> = service_impl;
```

Code I am trying to build is here:
https://github.com/WojciechMigda/cib-examples/blob/main/ticker-extended/src/main.cpp

Thanks!

--- a/include/cib/detail/find.hpp
+++ b/include/cib/detail/find.hpp
@@ -23,38 +23,27 @@ namespace cib::detail {
         }
     };
 
-    // clang-8 and below don't have constexpr find_first_of on std::string_view
-    CIB_CONSTEVAL std::string_view::size_type find_first_of(std::string_view str, char c) {
-        using size_type = std::string_view::size_type;
-        for (size_type i = 0; i < str.size(); i++) {
-            if (str[i] == c) {
-                return i;
-            }
-        }
-
-        return 0;
-    }
-
-    // clang-8 and below don't have constexpr find_last_of on std::string_view
-    CIB_CONSTEVAL std::string_view::size_type find_last_of(std::string_view str, char c) {
-        using size_type = std::string_view::size_type;
-        size_type match = str.size();
-        for (size_type i = 0; i < str.size(); i++) {
-            if (str[i] == c) {
-                match = i;
-            }
-        }
-
-        return match;
-    }
-
     template<typename Tag>
-    CIB_CONSTEVAL static std::string_view name() {
-        constexpr std::string_view function_name = __PRETTY_FUNCTION__;
-        constexpr auto lhs = find_first_of(function_name, '<');
-        constexpr auto rhs = find_last_of(function_name, '>');
-        constexpr auto service_name = function_name.substr(lhs, rhs - lhs);
-        return service_name;
+    constexpr static std::string_view name() {
+        #if defined(__clang__)
+            constexpr std::string_view function_name = __PRETTY_FUNCTION__;
+            constexpr auto lhs = 44;
+            constexpr auto rhs = function_name.size() - 2;
+
+        #elif defined(__GNUC__) || defined(__GNUG__)
+            constexpr std::string_view function_name = __PRETTY_FUNCTION__;
+            constexpr auto lhs = 59;
+            constexpr auto rhs = function_name.size() - 51;
+
+        #elif #elif defined(_MSC_VER)
+            constexpr std::string_view function_name = __FUNCSIG__;
+            constexpr auto lhs = 0;
+            constexpr auto rhs = function_name.size();
+        #else
+            static_assert(false, "Unknown compiler, can't build type name.");
+        #endif
+
+        return function_name.substr(lhs, rhs - lhs + 1);
     }
 
 

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -6,6 +6,7 @@ add_executable(tests
     detail/type_list.cpp
     detail/type_pack_element.cpp
     detail/meta.cpp
+    detail/type_to_string_view.cpp
     builder_meta.cpp
     callback.cpp
     nexus.cpp

--- a/test/detail/type_to_string_view.cpp
+++ b/test/detail/type_to_string_view.cpp
@@ -1,0 +1,11 @@
+#include <cib/cib.hpp>
+
+#include <catch2/catch_test_macros.hpp>
+
+#include <string_view>
+
+
+TEST_CASE("ensure types are converted to full string", "[type_to_string_view]") {
+    //using std::string_view_literals::sv;
+    REQUIRE(cib::detail::name<struct ABCDEFG>() == "ABCDEFG");
+}

--- a/test/detail/type_to_string_view.cpp
+++ b/test/detail/type_to_string_view.cpp
@@ -4,8 +4,9 @@
 
 #include <string_view>
 
+struct ABCDEFG;
 
 TEST_CASE("ensure types are converted to full string", "[type_to_string_view]") {
     //using std::string_view_literals::sv;
-    REQUIRE(cib::detail::name<struct ABCDEFG>() == "ABCDEFG");
+    REQUIRE(cib::detail::name<ABCDEFG>() == "ABCDEFG");
 }

